### PR TITLE
Topic/optimization in machine learning

### LINKED ISSUE #969: [Development Call]: Optimization techniques in machine learning
### Contact Details

kheiry.sohooli@gmail.com

### Is your content request related to a problem you've encountered during your research process? Please describe.

Knowledge about optimizing the algorithm is essential for any algorithm. 
I am currently working on XGBoost building block. And I think a building block dedicated to the optimization algorithm would be helpful in understanding the XGBoost and any other machine learning algorithms. 

### Additional context

_No response_

### Would you like this to be a Building Block or a Tutorial?

Building Block

### Purpose

To design an effective and efficient machine learning solution, it is vital to have enough knowledge about the optimization techniques. So having a building bock for that could facilitate it for users of the Tilburg Science Hub platform.  

### Motivation

_No response_

### Suggested title

Optimization techniques in machine learning 

### Suggested keywords

_No response_

### Required tools

_No response_

### References

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by K-sohooli**
This article provides an introduction to common optimization algorithms in ML. In this article, I try to explain these algorithms in a simple language and emphasize their importance in training the model. Also, the suitable optimization for each cost function is briefly mentioned. these concepts are overlooked sometimes due to their mathematics and complex appearance. so I tried to explain the basic formulas behind it very simply.

### CODE REVIEW

File: content/topics/Analyze/machine-learning/supervised/Optimization_ML.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+- This algorithm is favored for optimization in deep learning due to its ability to balance speed and stability effectively.
+
+#### Disadvantages: 
+- Convergence of an error cannot be assured with certainty.
+- Manually regulate the learning rate when employing MGD. If the learning rate is set too low, the convergence rate may decrease. Conversely, if the learning rate is too high, it may lead to failure in getting an absolute global or local minimum. Thus, careful control of the learning rate is essential in MGD.
+
+The image below shows the distinctions among three variants of Gradient Descent.
+<div align="center">
+  <img src="../images/GD_variants.png" width="400">
+  <p><a href="https://sweta-nit.medium.com/batch-mini-batch-and-stochastic-gradient-descent-e9bc4cacd461">Source</a></p>
```
Comment (K-sohooli) [inline content/topics/Analyze/machine-learning/supervised/Optimization_ML.md:105]: line 105 must be removed. 

I replaced the image from Medium with a new image I created in Python. so there's no need for this source anymore.

**Comment by K-sohooli** [comment]
Hello @lachlandeer , 
It's been a couple of months since I submitted this pull request, and I understand you may be busy with other tasks. I wanted to kindly remind you about it in case it got lost among other PRs.

**Comment by alexandervossen** [comment]
Hi @K-sohooli I had a look at it - albeit not being an expert :) So one thing I found a bit funny was the paragraph "Gradient descent algorithm in simple language" that begins with "Let us consider Equation (1)" :) That is not what I would deem simple language :) For readers like me a really simple example would be great! Could you incorporate that and use it throughout the post? Thanks! A

**Comment by lachlandeer** [comment]
Thanks @alexandervossen ... 

@srosh2000  can you also look this over and suggest any changes you have in mind?

**Comment by K-sohooli** [comment]
Hello @lachlandeer and  @alexandervossen , thanks for the feedback.
I updated the article by including a simple example in the session "Gradient descent algorithm in simple language" to make it easier to understand. However, Gradient descent is inherently a mathematical concept. So, it contains a lot of mathematical formulas. I only mention the basic formulas which are fundamental and necessary to explain the concepts. Therefore, having a few number of formulas is unavoidable in this article. Otherwise, it is impossible to explain the concepts. I have explained in simple language how Gradient descent helps the machine learning algorithms to be optimized, and I address the important parameters in the formula that we need to tune during the optimization of an ML algorithm. 
Thanks for your time.

**Comment by alexandervossen** [comment]
@K-sohooli I understand. But most of our methodological content is "inherently mathematical." We should nevertheless always strife to make it as accessible as possible.

**Comment by alexandervossen** [comment]
For me it looks fine @srosh2000 what do you think? I would love to have a blank pull request sheet for @lachlandeer so he can enjoy his trip home :)

**Comment by lachlandeer** [comment]
Looks fine to me. 

Probably too technical. But something we could revist later if we looked at the ML content collectively and wanted to bring it down one level technically

Thanks @K-sohooli
